### DISCUSSION TITLE
Is there a full description of the query language?

### DISCUSSION THREAD

**Discussion by haochenx** [PostgREST/postgrest#2652] [Q&A] [answered]
Is there a full description (or paper, etc.) of the query language accepted by PostgREST?

There is good documentation at https://postgrest.org/en/stable/api.html that describes how queries could be formed in a cookbook style, but clearly a DSL is indeed there. Before digging into the source code, is there a document/paper that describe the design of the full query language?

**Comment by steve-chavez** [comment, marked as answer]
Not yet, good idea for the docs.

This is outdated, but could help: https://github.com/PostgREST/postgrest-docs/issues/228#issuecomment-346981443

**Reply by haochenx** [reply]
Many thanks for the reply. If I'd like to start reading the source code for learning the grammar, which file would you suggest me to start? It seems that the [Parsers.hs](https://github.com/begriffs/postgrest/blob/cf19ad036926b31f9982e80f79a14e7cd93eaba3/src/PostgREST/Parsers.hs) mentioned in https://github.com/PostgREST/postgrest-docs/issues/228#issuecomment-345378902 is no longer there at the newest main

**Reply by steve-chavez** [reply]
Ah, the parsers are here now https://github.com/PostgREST/postgrest/blob/main/src/PostgREST/ApiRequest/QueryParams.hs

They're on a better state now as they have doctests, good for starting up.
